src/server/system_services/pool_server.js

```diff
@@ -88,6 +88,14 @@ function set_pool_controller_factory(pool_controller_factory) {
 
 }
 
+// The function checks whether the pool/resource has an owner
+// and only allows deletion in case that the owner is also the requester of the deletion
+function check_deletion_ownership(req, resource_owner_id) {
```

Comment:
If I recall correctly, the plan was to add the `owner_id` to the schema and not enforce any access control at this stage. Did anything in the plan change? 
I don't mind adding it now unless there are other concerns. In any case, we should probably at least have a flag in `config.js` to toggle it, which defaults to false (don't check). 

@nimrod-becker @liranmauda?

Reply:
I agree with having a flag on the `config.js`
I do think that the ask was to enforce it.
@nimrod-becker ?

Reply:
I think we said in the meeting that we will set only for remote and then check only if exists.
However after a conversation with Danny, I think we can go with setting always and checking according to config.js (which actually means an env set to replace if needed)